docs/book/src/development/development.md

```diff
@@ -342,6 +342,86 @@ $ sudo ln -s /run/user/$(id -u)/podman/podman.sock /var/run/docker.sock
 [hack-ci-gce-project]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/ci/gce-project.sh
 [hack-ci-openstack]: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/hack/ci/openstack.sh
 
+## Cluster Creation with ClusterClass
+
+This section provides a guide to create a Kubernetes cluster using ClusterClass. It covers the process of setting up environment variables, creating a Kind cluster, configuring a secret, applying ClusterClass and deploying the cluster.
+
+### Setting Environment Variables
+
+These environment variables are critical for configuring your Kubernetes cluster on OpenStack. cluster_name identifies your cluster within the infrastructure. openstack_ssh_key_name is added to the authorized keys on the nodes and bastion. kubernetes_version selects the pre-existing image used for the nodes. openstack_control_plane_machine_flavor (optional) defines the resource allocation for control plane machines. openstack_node_machine_flavor (optional) sets the resource specifications for worker nodes. openstack_cloud (optional) determines the OpenStack cloud environment for deploying your cluster.
+
+
+```bash
+# Define the name of your cluster
+export CLUSTER_NAME="<cluster_name>"
+# Desired Kubernetes Version for Your Cluster (e.g., "v1.28.5")
+export KUBERNETES_VERSION="<kubernetes_version>"
+
+# [Optional] SSH Keypair Name for Instances in OpenStack (Default: "")
+export OPENSTACK_SSH_KEY_NAME="<openstack_keypair_name>"
+# [Optional] Control Plane Machine Flavor (Default: m1.medium)
+export OPENSTACK_CONTROL_PLANE_MACHINE_FLAVOR="<openstack_control_plane_machine_flavor>"
+# [Optional] Node Machine Flavor (Default: m1.small)
+export OPENSTACK_NODE_MACHINE_FLAVOR="<openstack_node_machine_flavor>"
+# [Optional] OpenStack Cloud Environment (Default: capo-e2e)
+export OPENSTACK_CLOUD="<openstack_cloud>"
+
+```
+
+### Creating a Kind Cluster
+
+Create a Kind cluster. This process involves the initialization of Cluster API providers for OpenStack.
+
+```bash
+kind create cluster
+export CLUSTER_TOPOLOGY=true
+clusterctl init --infrastructure openstack
+```
+
+### Secret Configuration
+
+CAPO needs a clouds.yaml file in order to manage the OpenStack resources needed for the Cluster. This should be supplied as a secret named `${CLUSTER_NAME}-clouds-yaml`. You can create this secret for example with:
+
+```bash
+kubectl create secret generic ${CLUSTER_NAME}-clouds-yaml --from-file=clouds.yaml
+```
+
+### Applying ClusterClass and Cluster Configuration
+
+Use `clusterctl` to substitute variables in your ClusterClass configuration file and apply it to set up the cluster class and cluster.
+
+```bash
+clusterctl generate cluster ${CLUSTER_NAME} --from templates/clusterclass-dev-test.yaml  | kubectl apply -f -
+clusterctl generate cluster ${CLUSTER_NAME} --from templates/cluster-template-development.yaml  | kubectl apply -f -
+```
+
+#### Using Tilt with CAPO
```

Comment:
Merge this with https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/e89ec9ce14067866bab80c84da5060980a378f36/docs/book/src/development/development.md?plain=1#L53
I missed that we already had a link to it :facepalm: